test/app/model/ChessEngine.spec.ts

```diff
@@ -0,0 +1,87 @@
+
+    expect(engine.move(player, squareFrom, squareTo)).toIncludeSameMembers([
+      {
+        eventType: 'PieceWasMoved',
+        piece: whitePiece,
+        from: squareFrom,
+        to: squareTo,
+      },
+    ]);
+  });
+
+  it('Given white piece on A2 and black piece on A4, when move white piece from A2 to A4, then white piece was moved from A2 to A4 and piece from A4 was captured', () => {
+    const whitePiece = new Queen(Side.WHITE);
+    const blackPiece = new Pawn(Side.BLACK);
+    const boardWithPieces: SquareWithPiece = { A2: whitePiece, A4: blackPiece };
+    const chessBoard = new ChessBoard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const player = new Player(Side.WHITE);
+    const squareFrom: Square = { column: 'A', row: 2 };
+    const squareTo: Square = { column: 'A', row: 4 };
+
+    expect(engine.move(player, squareFrom, squareTo)).toIncludeSameMembers([
+      { eventType: 'PieceWasCaptured', piece: blackPiece, onSquare: squareTo },
+      { eventType: 'PieceWasMoved', piece: whitePiece, from: squareFrom, to: squareTo },
+    ]);
+  });
+
+  it('Should throw an Error if player wants to move oponents piece', () => {
+    const whitePiece = new Queen(Side.WHITE);
+    const blackPiece = new Pawn(Side.BLACK);
+    const boardWithPieces: SquareWithPiece = { A2: blackPiece, A4: whitePiece };
+    const chessBoard = new ChessBoard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const player = new Player(Side.WHITE);
+    const squareFrom: Square = { column: 'A', row: 2 };
+    const squareTo: Square = { column: 'A', row: 4 };
+
+    expect(() => engine.move(player, squareFrom, squareTo)).toThrowError('Player can not move other players pieces.');
+  });
+
+  it('Should throw an Error if player wants to move piece to not available square', () => {
+    const whitePiece = new Pawn(Side.WHITE);
+    const boardWithPieces: SquareWithPiece = { A2: whitePiece };
+    const chessBoard = new ChessBoard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const player = new Player(Side.WHITE);
+    const squareFrom: Square = { column: 'A', row: 2 };
+    const squareTo: Square = { column: 'A', row: 5 };
+
+    expect(() => engine.move(player, squareFrom, squareTo)).toThrowError('Piece can not move to given square.');
+  });
+
+  it('Should throw an Error if player wants to move twice', () => {
+    const whitePiece = new Pawn(Side.WHITE);
+    const boardWithPieces: SquareWithPiece = { A2: whitePiece };
+    const chessBoard = new ChessBoard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const player = new Player(Side.WHITE);
+    const squareStart: Square = { column: 'A', row: 2 };
+    const squareMiddle: Square = { column: 'A', row: 3 };
+    const squareFinish: Square = { column: 'A', row: 4 };
+
+    engine.move(player, squareStart, squareMiddle);
+
+    expect(() => engine.move(player, squareMiddle, squareFinish)).toThrowError();
```

Comment:
Aaa, dobra ten test jest :) Ale to mnie zmyliło. OK. Dziwne jest to radzenie sobie z sprawdzaniem wrzucania wyjatku, ale inaczej sie chyba nie da :) Jedynie co dodaj tutaj error message jak poprzednio.